File: scripts/ascii_fix_r.py
"""Replace non-ASCII punctuation in R sources for portable R CMD check."""
from pathlib import Path

REPLACEMENTS = {
    "\u2014": "-",  # em dash
    "\u2013": "-",  # en dash
    "\u2192": "->",
    "\u2265": ">=",
    "\u03b2": "beta",
    "\u2229": "intersect",
    "\u2713": "OK",
    "\u2717": "X",
    # mojibake from UTF-8 misread as Latin-1
    "\u00d4\u00c7\u00d6": "-",  # ÔÇö
    "\u00d4\u00e5\u00d2": "->",  # ÔåÆ
    "\u00d4\u00f6\u00c7": "-",  # ÔöÇ (box drawing)
}

r_dir = Path(__file__).resolve().parents[2] / "R"
for path in sorted(r_dir.glob("*.R")):
    text = path.read_text(encoding="utf-8")
    orig = text
    for old, new in REPLACEMENTS.items():
        text = text.replace(old, new)
    if text != orig:
        path.write_text(text, encoding="utf-8", newline="\n")
        print(f"fixed: {path.name}")

File: scripts/ascii_fix_r.R
# Replace non-ASCII punctuation in R sources (R CMD check portable packages).
files <- list.files("R", pattern = "\\.R$", full.names = TRUE)
replacements <- c(
  "\u2500" = "-",
  "\u2014" = "-",
  "\u2013" = "-",
  "\u2192" = "->",
  "\u2265" = ">=",
  "\u00b2" = "^2",
  "\u2026" = "...",
  "\u03b2" = "beta",
  "\u2229" = "intersect",
  "\u2713" = "OK",
  "\u2717" = "X",
  "\u00d4\u00c7\u00f6" = "-",
  "\u00d4\u00e5\u00c6" = "->",
  "\u00d4\u00f6\u00c7" = "-",
  "\u00d4\u00c7\u00aa" = "..."
)
for (f in files) {
  x <- readLines(f, warn = FALSE, encoding = "UTF-8")
  orig <- paste(x, collapse = "\n")
  for (i in seq_along(replacements)) {
    x <- gsub(names(replacements)[i], replacements[i], x, fixed = TRUE)
  }
  if (!identical(orig, paste(x, collapse = "\n"))) {
    writeLines(x, f, useBytes = FALSE)
    message("fixed: ", f)
  }
}
